### PR TITLE
Cable Spawners fixbatch 2: colour cables

### PR DESCRIPTION
**Pull request by TheTyrant25**
[BUG] [MINOR] [CLEANLINESS] [PERF] [MAPPING] [OBJECTS]
## About the PR 
So I tried out the multicoloured cable spawners and much to my surprise they completely didn't work at all. This PR fixes everything that was wrong, including and not limited to:
- coloured cablespawners connecting to each other
- Them not being able to detect anything already spawned in cables
- Not being able to detect anything already spawned in the southeast and southwest
- node cablespawners not working right
- red (parent) cablespawners ruining everything and connecting to everything.
- different coloured cables having bugged rubber names

I'm much better at coding now, and some of the code made me a little ashamed. Which is good. Life is growth.
Bonus: made `/obj/disposalpipespawners` abstract while i was fiddling around with making the parent cablespawner abstract. I reverted that eventually, but the disposalpipespawners were always supposed to be like that, i just ~~didn't know about abstract types yet~~ forgot to code them at the time.

## Why's this needed? 
Cable spawners were unusable, now they're fully functional.
In addition, I made some minor tweaks to make them initialise faster and whatnot.
For clarification, they worked fine up until now, but only if you used red exclusively. And no nodes. And no different colours, and certainly no crossing colours.
But hey, now you can place multiple (differently coloured) cable spawners in the same tile, which is neato (and allows for two coloured networks to actually connect at a knot).

### CODE REVIEW

File: code/modules/power/cable.dm [hunk 1]
```diff
@@ -415,6 +415,11 @@
 
 	return
 
+#define CHECK_AHEAD_LIST list(NORTH, EAST, NORTHWEST, NORTHEAST)
```
Comment (ZeWaka) [inline code/modules/power/cable.dm:418]: this is literally only used once, why is it a define

File: code/modules/power/cable.dm [hunk 2]
```diff
@@ -431,32 +436,31 @@
 	/// cable_surr uses the unique ordinal dirs to save directions as it needs to store up to 8 at once
 	var/cable_surr = 0
 
+	/// reinforced, thick cables. They should also connect to the regular kind.
+	node
```
Comment (ZeWaka) [inline code/modules/power/cable.dm:440]: please actually path these instead of relative pathing

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/power/cable.dm (hunk 1 above)
Comment (TheTyrant25) [reply code/modules/power/cable.dm:418]: ah, lol
i thought it'd be used more but i was wrong

File: code/modules/power/cable.dm (hunk 2 above)
Comment (TheTyrant25) [reply code/modules/power/cable.dm:440]: i've now pathed the node spawners, reinforced spawners and reinforced node spawners

**Review by ZeWaka — APPROVED** [review]

File: code/modules/power/cable.dm [hunk 3]
```diff
@@ -431,19 +431,11 @@
...
-/obj/cablespawner/New()
-	..()
-	if(current_state >= GAME_STATE_WORLD_INIT && !src.disposed)
-		SPAWN(1 SECONDS)
-			if(!src.disposed)
-				initialize()
-
 /obj/cablespawner/node
-	name = "node cable spawner"
+	name = "node reinforced cable spawner"
```
Comment (TobleroneSwordfish) [inline code/modules/power/cable.dm:435]:
```suggestion
	name = "node cable spawner"
```
?

**Pull request merged**